crp_APL/ctrl_vehicle_control/src/compensatory_model/compensatory_model.cpp

```diff
@@ -0,0 +1,235 @@
+            // finally, lateral feedforward acceleration is calculated - this is done to be in harmony 
+            // with the feedback path later (out of scope of this method)
+            m_targetAccelerationFF = m_targetCurvature*std::pow(input.vxEgo,2);
+            if(params.debugKPIs){printf("ayTarFF %f\n", m_targetAccelerationFF);}
+        }
+
+        void compensatoryModel::calculateFeedback(const controlInput& input, const controlParams& params)
+        {
+            // this method calculates the feedback target steering angle.
+            // First, control error is defined. A separate FB look ahead distance is calculated for this.
+            // orientation and position error are calculated.
+            // Secondly, the integral and derival parts are calculated.
+            // Third, the orientation error is calculated.
+            // Fourth, the errors are weighted and summed with the corresponding control coefficients.
+            // Finally, lateral acceleration target is mapped to steering angle target.
+
+            m_lookAheadDistanceFb = std::max(params.fbMinLookAheadDistance, params.fbLookAheadTime*input.vxEgo);
+
+            m_posErr = 
+                m_coefficients[0]+
+                m_coefficients[1]*m_lookAheadDistanceFb +
+                m_coefficients[2]*std::pow(m_lookAheadDistanceFb,2)+
+                m_coefficients[3]*std::pow(m_lookAheadDistanceFb,3);
+
+            m_orientationErr = atan(
+                m_coefficients[1]+
+                2.0f*m_coefficients[2]*m_lookAheadDistanceFb+
+                3.0f*m_coefficients[3]*std::pow(m_lookAheadDistanceFb,2));
+
+            m_posIntegralError = m_posIntegralError+m_posErr*params.dT;
+
+            m_posDerivativeError = (m_posErr-m_posErrPrev)/params.dT;
+            m_posDerivativeError = m_posDerivativeFilter.lowPassFilter(m_posDerivativeError, m_posDerivativeError_prev, 0.9f);
+
+            m_targetAccelerationFB = params.fbPGain*m_posErr +
+                params.fbDGain*m_posDerivativeError;
+
+            if (m_posIntegralError < params.fbIntegralLimit)
+            {
+                m_targetAccelerationFB = m_targetAccelerationFB +
+                params.fbIGain*m_posIntegralError;
+            }
+            
+            m_targetAccelerationFB = m_targetAccelerationFB + params.fbThetaGain * m_orientationErr;
+            
+            if(params.debugKPIs){printf("errors %f, %f, %f\n", m_posErr, m_posIntegralError, m_posDerivativeError); printf("ayTarFB %f\n", m_targetAccelerationFB);}
+
+            m_posErrPrev = m_posErr;
+        }
+
+        void compensatoryModel::cutRelevantLocalSnippet()
+        {
+            // this method cuts a relevant length of the total trajectory, based on the complete 
+            // path transformed to the ego coordinate frame
+            // step 1: searching for nearest point in trajectory
+            unsigned long int startIdx = -1;
+            unsigned long int stopIdx = -1;
```

Comment:
Consider changing this, it looks a bit weird for me.
```suggestion
            unsigned long int stopIdx = std::numeric_limits<unsigned long int>::max();;
```

Reply:
max -ing out this is a bit risky, if the stop idx remains the same the indexing would run out